packages/lit-html/src/directives/class-map.ts

```diff
@@ -76,7 +90,10 @@ class ClassMapDirective extends Directive {
       // We explicitly want a loose truthy check of `value` because it seems
       // more convenient that '' and 0 are skipped.
       const value = !!classInfo[name];
-      if (value !== this._previousClasses.has(name)) {
+      if (
+        value !== this._previousClasses.has(name) &&
+        !this._staticClasses?.has(name)
```

Comment:
This is adding runtime overhead to what is potentially a hot-ish path. It doesn't appear to show up on existing benchmarks so seems ok.